### DISCUSSION TITLE
Docker image latest tag should point to latest release instead of main branch

### DISCUSSION THREAD

**Discussion by acidghost** [open-webui/open-webui#3125] [General]
Tools like [diun](https://github.com/crazy-max/diun) can be used to watch for updates to Docker images.
The way it works is via container tags: deploy an image like `image:tag` and you'll get notified of new versions of that tag.

Usually by watching the `latest` tag you get updates for _released_ versions. This is not the case for this project, the `main` and `latest` tags are the same.

ATM I could watch for a specific version, e.g. `0.3`, and get updates for any `0.3.x`, but I would not get updates for any `0.4.x`.

IMO the `latest` tag should point to the latest release and the `main` and `dev` tags can point to the respective branches for people who want that.

**Comment by acidghost** [comment]
@tjbck Why move it to a discussion? Can't it be discussed in the issue?

**Reply by justinh-rahb** [reply]
It's not an issue. `main` **is** the release branch. Tags are set as we intend them.

**Reply by acidghost** [reply]
Why having both a `main` and `latest` tag that point to the same thing? This just creates confusion..

To add to the confusion the latest release on GitHub is the numbered one, not some snapshot of the `main` branch (https://github.com/open-webui/open-webui/releases/latest). Look at something like neovim to implement nightly versions (https://github.com/neovim/neovim/releases).

What's the point of having releases at all if you say that *the* release is a rolling release? Not even a rolling release, it just gets pushed multiple times a day.

This is just an inconvenience to users who want to keep up with the updates but don't want to deploy the bleeding edge.

Can you please point out how does a user get the latest *numbered* release in stuff like Diun, Watchtower, etc?

**Reply by justinh-rahb** [reply]
> Why having both a `main` and `latest` tag that point to the same thing? This just creates confusion..

This is common practice.

**Reply by acidghost** [reply]
Can you please point to some example?

**Reply by cheahjs** [reply]
The current development practice on Open WebUI is that any commit to `main` is considered a release, which may or may not be accompanied by a tag. The bleeding edge is following the `dev` branch.

**Reply by acidghost** [reply]
Thanks for the explanation.

@justinh-rahb I was just wondering about a problem I encountered, I hope I didn't came out as pretending something. No need to be this passive-aggressive 🙄

**Reply by tjbck** [reply]
![shrek-shrek-rizz](https://github.com/open-webui/open-webui/assets/25473318/c676c7cd-f2af-4179-8db3-678f60b80f0e)
